doc/adv-manual/pandas.rst

```diff
@@ -40,3 +40,50 @@ you can get a slice of hazard curves by using the ``.sel`` method:
 Notice that the `.sel` method does not reduce the number of dimensions
 of the original array (4 in this case), it just reduces the number of elements.
 It was inspired by a similar functionality in xarray.
+
+Example: how many events per magnitude?
+---------------------------------------
+
+When analyzing an event based calculation, users are often interested in
+checking the magnitude-frequency distribution, i.e. to count how many
+events of a given magnitude present in the stochastic event set for
+a fixed investigation time and a fixed ``ses_per_logic_tree_path``.
+You can do that with code like the following:
+
+.. code-block:: python
+                
+ def print_events_by_mag(calc_id):
+     # open the DataStore for the current calculation
+     dstore = datastore.read(calc_id)
+     # read the events table as a Pandas dataset indexed by the event ID
+     events = dstore.read_df('events', 'id')
+     # find the magnitude of each event by looking at the 'ruptures' table
+     events['mag'] = dstore['ruptures']['mag'][events['rup_id']]
+     # group the events by magnitude
+     for mag, grp in events.groupby(['mag']):
+         print(mag, len(grp))   # number of events per group
+
+If you want to now the number of events per realization and per stochastic
+event set you can just refine the `groupby` clause, using the list
+``['mag', 'rlz_id', 'ses_id']`` instead of simply ``['mag']``.
+
+Given an event, it is trivial to extract the ground motion field
+generated by that event, if it has been stored (warning: events
+producing zero ground motion are not stored). It is enough to read
+the ``gmf_data`` table indexed by event ID, i.e. the ``eid`` field:
+
+.. code-block:: python
+
+   >>> eid = 20  # consider event with ID 20
+   >>> gmf_data = dstore.read_df('gmf_data/data', 'eid')
+   >>> gmf_data.loc[eid]
+         sid     gmv_0
+   eid               
+   20    93   0.113241
+   20   102   0.114756
+   20   121   0.242828
+   20   142   0.111506
+
+The ``gmv_0`` refers to the first IMT; here I have shown an example with a
+single IMT, in presence of multiple IMTs you would see multiple columns
+``gmv_0, gmv_1m, gmv_2, ...``. The ``sid`` column refers to the site ID.
```

Comment:
You might want to add the following: As a following step, you can reorder these values of shaking (for a single IMT) and compute the hazard curve at this site.